ir/README.md

```diff
@@ -0,0 +1,33 @@
+# Intermediate Representation (IR)
+
+This crate contains the intermediate representation for AirScript, `AirIR`.
+
+The purpose of the `AirIR` is to provide a simple and accurate representation of an AIR that allows for optimization and translation to constraint evaluator code in a variety of target languages.
+
+## Generating the AirIR
+
+Generate an `AirIR` from an AirScript AST (the output of the AirScript parser) using the `from_source` method. `from_source` will return a new `AirIR` or an `Error` of type `SemanticError` if it encounters any errors while processing the AST.
```

Comment:
Minor nit: I'm wondering if the start of sentence can be with `from_source` here and below or if we should change it to something like "The `from_source` method will return a new". It's a little more verbose but may be cleaner to look at.